spirv_msl.cpp

```diff
@@ -1033,10 +1083,14 @@ void CompilerMSL::preprocess_op_codes()
 
 	suppress_missing_prototypes = preproc.suppress_missing_prototypes;
 
+    // Disable warning about missing braces for array<T> template to make arrays a value type
+    add_pragma_line("#pragma clang diagnostic ignored \"-Wmissing-braces\"");
```

Comment:
Can this be added conditionally to avoid getting changes for 700++ files in this PR?

Reply:
Good point.
I moved that code to `CompilerMSL::emit_header()` where `-Wmissing-prototypes` is emitted, too. At that point we can check if `SPVFuncImplUnsafeArray` is contained in `spv_function_implementations`.

Reply:
The first update reduced the amount of changed files to 436